### DISCUSSION TITLE
differenet behaviours between localhost and github

### DISCUSSION THREAD

**Discussion by Freed-Wu** [jekyll/jekyll#9341] [Q&A] [answered]
I create a blog by jekyll. However, the `_site` built by jekyll has different behaviours between localhost and github.

```shell
git clone --branch=gh-pages https://github.com/Freed-Wu/Freed-Wu.github.io _site
python -m http.server -d _site
```

```shell
xdg-open http://localhost:8000/2023/
```

The web page exists. `_site/sitemap.xml` contains the url, too.

However, in github,

```shell
xdg-open https://Freed-Wu.github.io/2023/
```

The web page doesn't exist.

`wget https://Freed-Wu.github.io/sitemap.xml` doesn't contain the url.

What happened?

**Comment by ashmaroli** [comment, marked as answer]
The reason behind this is that you have set up GitHub Pages settings incorrectly.
You have correctly set up GitHub Actions to build your site with Jekyll 4 and upload it to the `gh-pages` branch. But unfortunately because of incorrect GitHub settings, GitHub rebuilds the site using Jekyll 3.9 (classic GitHub Pages) and deploys that to your domain.
This can be confirmed by `View Page Source` of your homepage. You will see `<meta name="generator" content="Jekyll v3.9.3" />`
